lib/hotel.rb

```diff
@@ -0,0 +1,196 @@
+require 'date'
+require_relative 'room.rb'
+require_relative 'reservations.rb'
+require_relative 'block.rb'
+require 'pry'
+
+module Hotel_System
+
+  class Hotel
+
+    attr_reader :all_rooms, :room_list, :list_of_rooms, :room_number, :reservation
+
+    attr_accessor :all_reservations, :room_object, :room_price, :available_rooms_hash, :avail_check_by_date, :available_room_list, :master_list, :block_rooms, :list_of_room_numbers_in_block
+
+    def initialize(num_of_rooms)
+      @all_rooms = fill_hotel(num_of_rooms)
+      @all_reservations = []
+    end
+
+    def return_room_object_by_num(room_number)
+      room_object = @all_rooms[room_number - 1]
+    end
+
+    def list_of_rooms
+      room_list = []
+      @all_rooms.each do |room|
+        room_list << room.room_number
+      end
+      return room_list
+    end
+
+    def make_reservation(room_number, check_in, check_out)
```

Comment:
I don't see any checks to ensure that the room hasn't already been reserved for the given date.  If this method is not to be called from outside the class, it should be `private`.